eng/packages/http-client-csharp-mgmt/generator/Azure.Generator.Management/src/Providers/OperationMethodProviders/ResourceOperationMethodProvider.cs

```diff
@@ -54,42 +55,76 @@ internal class ResourceOperationMethodProvider
             RestClientInfo restClientInfo,
             InputServiceMethod method,
             MethodProvider convenienceMethod,
             bool isAsync,
-            string? methodName = null)
+            string? methodName = null,
+            FormattableString? description = null,
+            bool forceLro = false)
         {
             _enclosingType = enclosingType;
             _contextualPath = contextualPath;
-            _resource = InitializeResource(enclosingType);
             _restClient = restClientInfo.RestClientProvider;
             _serviceMethod = method;
             _convenienceMethod = convenienceMethod;
             _isAsync = isAsync;
+            InitializeLroFlags(
+                _serviceMethod,
+                forceLro: forceLro,
+                ref _isLongRunningOperation,
+                ref _isFakeLongRunningOperation);
             _methodName = methodName ?? convenienceMethod.Signature.Name;
-            _responseGenericType = _serviceMethod.GetResponseBodyType();
-            _isGeneric = _responseGenericType != null;
-            _isLongRunningOperation = _serviceMethod.IsLongRunningOperation();
-            _isFakeLongRunningOperation = _serviceMethod.IsFakeLongRunningOperation();
+            _description = description ?? convenienceMethod.Signature.Description;
+            InitializeTypeInfo(
+                _serviceMethod,
+                ref _originalBodyType,
+                ref _returnBodyType,
+                ref _resourceClient);
             _clientDiagnosticsField = restClientInfo.DiagnosticsField;
             _restClientField = restClientInfo.RestClientField;
             _signature = CreateSignature();
             _bodyStatements = BuildBodyStatements();
         }
 
-        // TODO -- we should not need to do this. We need the resource only because sometimes we need to wrap the return type into its corresponding resource.
-        // We should have a cache in the outputlibrary for it.
-        private static ResourceClientProvider InitializeResource(TypeProvider enclosingType)
-            => enclosingType switch
+        private void InitializeLroFlags(
+            in InputServiceMethod serviceMethod,
+            in bool forceLro,
+            ref bool isLongRunningOperation,
+            ref bool isFakeLongRunningOperation)
+        {
+            isLongRunningOperation = serviceMethod.IsLongRunningOperation();
+            // when the method is not a real LRO, but we have to force it to be an LRO, the fake lro flag is true.
+            isFakeLongRunningOperation = forceLro && !isLongRunningOperation;
+        }
+
+        private void InitializeTypeInfo(
+            in InputServiceMethod serviceMethod,
+            ref CSharpType? originalBodyType,
+            ref CSharpType? returnBodyType,
+            ref ResourceClientProvider? wrappedResourceClient)
+        {
+            originalBodyType = serviceMethod.GetResponseBodyType();
```

Comment:
```suggestion
            originalBodyType = _serviceMethod.GetResponseBodyType();
```
the input parameter can be simplified

Reply:
I forgot to add the `static` modifier to this method.
I am using an input parameter because I tried to make this independent from the order of statements in the ctor.
I once fell into the pit hole.
I believe static initializer should be a better practice